### DISCUSSION TITLE
Getting and viewing Static IP

### DISCUSSION THREAD

**Discussion by MJohnson459** [aws/copilot-cli#4835] [Q&A] [answered]
There are a few questions already on static IPs (#1914 ) that say it is supported but stop the explanation once a NAT has been set up. In my service manifest I have this block as suggested but I don't see how to get the static IP.

```yaml
network:
  vpc:
    placement: private
```

I can see that two NAT gateways are created `copilot-{app}-test-0` and `copilot-{app}-test-0` each with a "Primary public IPv4 address", are these the egress IPs? If so, are they static or will they be reset if the app is deployed to again?

**Comment by efekarakus** [comment, marked as answer]
Hi @MJohnson459 👋 

> "Primary public IPv4 address", are these the egress IPs

Yeah, these are the egress IPs 👍 

>  If so, are they static or will they be reset if the app is deployed to again?

They are static! They will _not_ get replaced when you deploy the services or jobs again. The NAT gateways are created in the environment stack and get shared across all your services deployed in the environment.

**Reply by MJohnson459** [reply]
Thank you! Is there a way to view them from the copilot CLI?

**Reply by efekarakus** [reply]
oh that's such a great idea! There isn't a way to do, but we should definitely add that to `env  show` I created a feature request here: https://github.com/aws/copilot-cli/issues/4837
